### PR TITLE
expanded release docs

### PR DESCRIPTION
**Pull request by ckerr**
I'm attempting to make my first Electron release and have hit a couple of missing instructions and also some incorrect assumptions on my part. This branch expands the `release.md` documentation to fill in these gaps.

I'll remove the [WIP] status after the release is done, but welcome comments at any point.

Special thanks to @jkleinsc for a lot of help :)

### CODE REVIEW

File: docs/development/releasing.md [hunk 1]
```diff
@@ -201,8 +251,34 @@ the correct checksums as specified in the SHASUMS files.
 
 ## Publish to npm
 
-Once the publish is successful, run `npm run publish-to-npm` to publish to
-release to npm.
+Before publishing to npm, you'll need to log into npm as Electron. Optionally,
```
Comment (zeke) [inline docs/development/releasing.md]: log into npm as `electron`

File: docs/development/releasing.md [hunk 2]
```diff
@@ -201,8 +251,34 @@ the correct checksums as specified in the SHASUMS files.
...
+```
+
+Publish the release to npm.
+```sh
+$ npm whoami
+electron
+$ npm run publish-to-npm
+```
+
+Note: the `publish-to-npm` script may have trouble with Node 7 or higher. If
```
Comment (zeke) [inline docs/development/releasing.md]: Curious about this Node 7+ issue. What went wrong?

Comment (jkleinsc) [reply docs/development/releasing.md]: @zeke when I have run publish-to-npm in the past it fails on newer versions of npm (eg npm 5).

Here is the error:
```
```node script/publish-to-npm.js
npm ERR! Cannot read property 'algorithm' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jkleinsc/.npm/_logs/2017-08-30T20_50_53_319Z-debug.log
Error: Error: Command failed: npm publish /var/folders/yh/1n_9z9z97fn0tj_bpp3g9bqw0000gn/T/electron-npm117730-36976-1vxt06u.3cyk0dlsor/electron-1.7.7.tgz --tag beta
npm ERR! Cannot read property 'algorithm' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jkleinsc/.npm/_logs/2017-08-30T20_50_53_319Z-debug.log```
```

File: docs/development/releasing.md [hunk 3]
```diff
@@ -5,19 +5,50 @@ This document describes the process for releasing a new version of Electron.
...
+[Version Change Rules](../tutorial/electron-versioning.md#version-change-rules)
+for more information.
+
+**NB:** If releasing from a branch, e.g. 1-8-x, check out the branch with
+`git checkout 1-8-x` rather than `git checkout -b remotes/origin/1-8-x`.
+The scripts need `git rev-parse --abbrev-ref HEAD` to return a short name,
+e.g. no `remotes/origin/`
+
+## Set your tokens and environment variables
+The Electron S3 Bucket in LastPass has environment variables needed for the
```
Comment (jkleinsc) [inline docs/development/releasing.md]: @ckerr given that this is a public facing document, I don't think we should explicitly document GitHub process here.  Maybe a generic statement about you'll need Electron S3 credentials?

**Comment by ckerr** [comment]
Wow, I had forgotten this PR existed.

At first I was wondering if it was even worth keeping after all this time, but as I read it I realized that I'd forgotten half of these things -- so it's probably worth dusting off and trying to land again :smile:

**Comment by ckerr** [comment]
@jkleinsc bump :smile:

File: docs/development/releasing.md [hunk 4]
```diff
@@ -147,30 +178,56 @@ For a `minor` release, e.g. `1.8.0`, use this format:
...
+4. Click 'Save draft'. **Do not click 'Publish release'!**
+5. Wait for all builds to pass before proceeding.
+6. In the `release` branch, verify that the release's files have been created:
+```sh
+$ git rev-parse --abbrev-ref HEAD
+release
+$ npm run release -- --validateRelease
+```
 
+## Merge temporary branch
```
Comment (jkleinsc) [inline docs/development/releasing.md]: This is only needed for pre 2.0.0 branches.  2-0-x does not use a release branch and doesn't need to be merged.

File: docs/development/releasing.md [hunk 5]
```diff
@@ -185,8 +242,34 @@ the correct checksums as specified in the SHASUMS files.
...
+you may find [npmrc](https://www.npmjs.com/package/npmrc) to be a useful way
+to keep Electron's profile side-by-side with your own:
+```sh
+$ sudo npm install -g npmrc
+$ npmrc -c electron
+Removing old .npmrc (default)
+Activating .npmrc "electron"
+```
+
+The Electron account's credentials are stored in LastPass under the entry
```
Comment (jkleinsc) [inline docs/development/releasing.md]: We should eliminate the reference to LastPass for security reasons.

File: docs/development/releasing.md [hunk 6]
```diff
@@ -185,8 +242,34 @@ the correct checksums as specified in the SHASUMS files.
...
+```
+
+Publish the release to npm.
+```sh
+$ npm whoami
+electron
+$ npm run publish-to-npm
+```
+
+Note: the `publish-to-npm` script may have trouble with Node 7 or higher. If
```
Comment (jkleinsc) [inline docs/development/releasing.md]: In regards to this note, Node 8 works fine, but you should be using Node 8 throughout the whole process particularly in light of that other Node issue we had.

**Comment by ckerr** [comment]
@jkleinsc, I didn't see your responses to this -- I really should tend these old pulls more often! I've done some copyediting here to address your suggestions. Thanks!

**Pull request merged**